### DISCUSSION TITLE
[GPX output] distance difference with Huawei app

### DISCUSSION THREAD

**Discussion by phipad** [CTHRU/Huawei-TCX-Converter#67] [Q&A]
Hi CTHRU,
I'm a user of your python software for a couple of months now, and I have to admit that is  amazing job, that Huawei isn't able to provide for years now.
So, a big thanks for this !
My discussion is about the GPX output files, and especially the distance output.
While Huawei gives on the GT2 Pro watch and on the Huawei Helath app a distance, the output gives another one, usually shorter.
For example, during this day running, the watch and the app said 10.02 km distance while the Hitrava output gave 9.88 km.
Do you know what could explain this difference seen on my running activities ?
Thanks a lot once again for your job !

**Comment by CTHRU** [comment]
Hello, thanks for reaching out and thank you for your kind words.

I think your question relates to the same behavior another user noticed. You can find an explanation as to why this might happen in [issue 63](https://github.com/CTHRU/Hitrava/issues/63) which is still open.
I could think of a possible 'fix' that uses the distance and time data from the header in the Huawei ZIP/JSON data rather than the calculated data based on the detailed track information. However not sure at this time whether Strava could/would use that info in the converted file. To be continued.

**Reply by phipad** [reply]
Many thanks for your quick feedback.
It seems that in issue 63, the difference is on the total time and not the total distance.
On my side , I used the python program with --file option, and the Hitrack file taken from Bluestack soft. And the feedback in the GPX conversion message was a shorter distance.
What seems strange to me is that I'm sometimes running with another watch (Amazfit) at the same time, and the 2 watches give the same distance and same speed. Only the conversion changes the distance.
Anyway, I will look with deep interest your next releases, since you are the only one to give some hope to Huawei Watches owners who use Strava.

**Comment by CTHRU** [comment]
To give you a heads-up, I checked for a possibility to pass the global Huawei numbers in the converted file, but it won't be that straight forward. Whenever a pause or GPS loss occurs during an activity (happens more often than expected), The intermediate track/distance/time has to be used in separate parts of the converted data. If I would just 'fix' the last part by adding a final distance/time record, it will probably result in messy data at the end of the activity. I think for now I have to leave it as is. Maybe a better alternative will come along the way.

**Comment by phipad** [comment]
Hello Christoph,
many thanks for your analysis and reply.
It seems a real tricky point, adn I'm afraid Huawei won't help on this.
I will keep following you !

**Comment by smonov** [comment]
Hello,

I also faced the described issue. Since I know the real distance and time from the Huawei Health app is there a way to fix this issue manually in the converted tcx file, before uploading it in Strava? If yes which fields have to be edited?

Regards,

**Comment by CTHRU** [comment]
I might have found a 'fix' to get the distances in sync. If you want, you can try out the script in the [5.1.0 development branch ](https://github.com/CTHRU/Hitrava/tree/dev-5.1.0)

Your feedback whether this would be a viable solution is welcome.
